src/Oscoin/P2P/Types.hs

```diff
@@ -78,47 +205,19 @@ deriving instance Show (f (NodeId c)) => Show (NodeAddr f c)
 -- enclosed in square brackets as per <https://tools.ietf.org/html/rfc3986#section-3.2.2 RFC 3986, Section 3.2.2>
 -- in order to delimit it from the port number.
 --
--- Examples:
---
--- >>> readNodeAddr "[2001:db8:00:00:00:00:00:01]:42"
--- Right (NodeAddr {nodeId = Nothing, nodeHost = "2001:db8::1", nodePort = 42})
---
--- >>> readNodeAddr "[2001:db8::01]:42"
--- Right (NodeAddr {nodeId = Nothing, nodeHost = "2001:db8::1", nodePort = 42})
---
--- >>> readNodeAddr "127.0.0.1:42"
--- Right (NodeAddr {nodeId = Nothing, nodeHost = "127.0.0.1", nodePort = 42})
---
--- >>> readNodeAddr "goo.gl:42"
--- Right (NodeAddr {nodeId = Nothing, nodeHost = "goo.gl", nodePort = 42})
---
 readNodeAddr :: String -> Either String (NodeAddr Maybe c)
-readNodeAddr s = do
-    (h, more) <- (first show <$> readIP s) <|> readHost s
-    p         <- note "Invalid port number" (readMaybe more)
-    pure NodeAddr
-        { nodeId   = Nothing
-        , nodeHost = h
-        , nodePort = p
-        }
-  where
-    readIP :: String -> Either String (IP, String)
-    readIP ('[' : more) =
+readNodeAddr = \case
```

Comment:
Future self may also make the port number optional here. We would need a contant to determine the default value, though, not hide it in the option parser for `main`